src/vs/workbench/parts/extensions/browser/extensionsActions.ts

```diff
@@ -78,6 +80,14 @@ export class InstallAction extends Action {
 	}
 
 	run(): TPromise<any> {
+		if (this.extension.recommendationInfo) {
+			this.telemetryService.publicLog('extensionRecommendations:install', {
```

Comment:
Successful installation can be tracked in the `onDidInstallExtension` event in `ExtensionsWorkbenchService`, by which time the recommendedInfo is lost. So I was ok with tracking just the intent